### PR TITLE
Allow running tests on arm64 runners

### PR DESCRIPTION
**Pull request by pat-s**
builds upon https://github.com/woodpecker-ci/woodpecker/pull/2232

NB: the change `pats22/xgo:go-1.21.3` is only as long as https://github.com/techknowlogick/xgo/pull/224 is not yet merged and an arm image is available upstream.

### CODE REVIEW

**Comment by 6543** [comment]
Nice ... i still want to wait until xgo has the patches merged :)

**Comment by pat-s** [comment]
Sure. Let's see how upstream goes. But a PoC was needed anyhow :)

**Comment by pat-s** [comment]
@6543 even though CI is green, compile-server-preview fails but the failure is not reflected by the ci

File: .woodpecker/binaries.yml [hunk 1]
```diff
@@ -72,7 +72,6 @@ steps:
     commands:
       - ls -la dist/*.*
       - cat dist/checksums.txt
```
Comment (6543) [inline .woodpecker/binaries.yml:74]:
```suggestion
      - cat dist/checksums.txt

```

**Comment by pat-s** [comment]
@6543 Promising new image, all tests worked so far. Test failure in `binaries` unrelated.

**Comment by pat-s** [comment]
@6543 If you want to test with an arm runner again, it might be worth it. Unclear how long it will take until tech really has the time to test this upstream.

**Comment by 6543** [comment]
ok - let me deploy the autoscaler and then update this pull to let it run.

might take some time

**Comment by 6543** [comment]
xgo works :tada:  - nfpm fails :D

**Comment by pat-s** [comment]
> nfpm fails :D

That issue looks unrelated AFAICS - it's not a compile issue but a missing file.

The issue is `dist/server/linux/amd64/woodpecker-server"` vs `dist/server/linux_amd64/woodpecker-server"` in the output locations.

I couldn't find a run where `build-deb-rpm` succeeded lately, so it might be a general issue unrelated to this PR.

**Comment by pat-s** [comment]
Given that the `xgo` compiler seems working now, I think we should give it a go, i.e. deploy arm runners. There's a lot to gain/save and doing so might also help tech adding upstream support. What do you think?

**Comment by pat-s** [comment]
Updated images to 1.22.1. 

Feedback would be great ;)

**Comment by 6543** [comment]
it did run on x86 right now ... let me change the autoscaler to spawn arm ... not jet but soon ish

**Comment by 6543** [comment]
what we could do is to explizite lable workflows to use x86 and then cange it to arm in here ...

**Comment by pat-s** [comment]
> what we could do is to explizite lable workflows to use x86 and then cange it to arm in here ...

Certainly good for testing. But a bit of effort to do. Though I am meanwhile pretty sure the arm build fine.

**Comment by pat-s** [comment]
@6543 A bit hard to make some progress here. I am slowly loosing motivation a bit as I can't do anything on my side anymore besides asking/pinging for external action. both here and in `xgo`.

If this is not important enough, we can also close it and just stay on `amd64` (?).

**Comment by 6543** [comment]
the only thing blocking it is https://github.com/techknowlogick/xgo/pull/224

so we can just make the autoscaler more inteligent and deploy arm and only if needed by workflow (label selector) use x64

**Comment by pat-s** [comment]
> the only thing blocking it is https://github.com/techknowlogick/xgo/pull/224

is not blocking, there is a functional image built from this PR under my account `pats22` which is also referenced in this PR and tested to work in both ways.

**Comment by 6543** [comment]
well in this case we can just use that meanwhile ... just add a codecoment (todo: switch to main once X is merged...)

**Comment by pat-s** [comment]
> well in this case we can just use that meanwhile ... just add a codecoment (todo: switch to main once X is merged...)

This is what I am trying to express since the last five comments or so 😅

**Review by xoxys — COMMENTED** [review]
Some IDEs can create a tasks tree from `TODO:` commends.

**Review by xoxys — APPROVED** [review]

**Pull request merged**